src/main/java/io/beekeeper/pingdom/resources/PingdomHealthResource.java

```diff
@@ -103,6 +106,15 @@ public PingdomHealth get(@QueryParam("key") String key, @QueryParam("severity")
                                 .append('\n');
             }
         }
-        return new PingdomHealth(status.length() == 0 ? PingdomHealth.HEALTH_OK : status.toString(), timer.getSnapshot().getMean() / 1000000.0d);
+        String statusMessage;
+        if (status.length() == 0) {
+            statusMessage = PingdomHealth.HEALTH_OK;
+        } else {
+            statusMessage = status.toString();
+            if (log.isWarnEnabled()) {
```

Comment:
no need to make use of this if statement. This is only useful if the message generation was to be expensive. We already have the message here

Reply:
this is the last thing to fix!

Reply:
Fixed by a93efc5.